folium/features.py

```diff
@@ -1095,6 +1095,9 @@ def get_by_key(obj, key):
 
             def color_scale_fun(x):
                 key_of_x = get_by_key(x, key_on)
+                if not key_of_x:
+                    if len(key_on) > 0:
+                        raise("key_on value not found in geojson")
```

Comment:
I don't know if this is the right syntax.  I would suggest to raise it with `raise ValueError('my message')`. A value error, because the type of `key_on` is right (it's a string), but its value is wrong.